code/mission/missionmessage.cpp

```diff
@@ -2276,8 +2274,7 @@ void message_maybe_distort_text(char *text, int shipnum, bool for_death_scream)
 
 			++i;
 		}
-		Assertion(result_str.size() <= buffer_size, "Buffer after scrambling message is bigger than before!");
-		strcpy(text, result_str.c_str());
+		text = result_str;
```

Comment:
You could use `std::move` for a bit of efficiency:
```suggestion
		text = std::move(result_str);
```